### PR TITLE
Added welcome email ability.

### PR DESCRIPTION
**Pull request by theclayton**

### CODE REVIEW

File: include/routes/v1/auth.js [hunk 1]
```diff
@@ -133,12 +133,6 @@ var registerRoute = async (req, res) => {
   //     });
   //     return;
   // }
-  req.body.domain = req.params.domain || 'default';
```
Comment (Phara0h) [inline include/routes/v1/auth.js:136]: why was this removed?

File: include/routes/v1/auth.js [hunk 2]
```diff
@@ -159,19 +159,31 @@ var registerRoute = async (req, res) => {
...
+  if (config.registration.sendWelcomeEmail === true && email) {
+    var failed = await Email.sendWelcome(user);
+
+    if (failed !== false) {
+      config.log.logger.info(`Sent welcome email to: ${email}.`);
+      sentWelcomeEmail = true;
+    }
+  }
+
+  res.code(200).send(`Account Created${sentWelcomeEmail ? ', Sent Welcome Email' : ''}`);
```
Comment (Phara0h) [inline include/routes/v1/auth.js:186]: would go back to the old message of `Account Created` If a sent welcome email was sent you can log it with the config.log.logger.debug

**Review by Phara0h — CHANGES_REQUESTED** [review]

File: include/routes/v1/auth.js (hunk 1 above)
Comment (theclayton) [reply include/routes/v1/auth.js:136]: It seemed repetitive.

Further down, line 159 declares `var domain = 'default';`
lines 162-164 ```  if (req.params.domain) {
    domain = req.params.domain.toLowerCase();
  }```
